config/routes.rb

```diff
@@ -0,0 +1,75 @@
+Rails.application.routes.draw do
+  # The priority is based upon order of creation: first created -> highest priority.
+  # See how all your routes lay out with "rake routes".
+  root 'markets#index'
+
+  get "products/new/:id" => "products#new", as: "new_product"
+
+  get "sales/new/:id" => "sales#new", as: "new_sale"
+
+  get "vendors/new/:id" => "vendors#new", as: "new_vendor"
+
+  get "sales/:id/:name" => "sales#month_sale", as: "month_sale"
+
+  get "markets/1/:letter" => "markets#lettermarket", as: "letter_market"
+
+  get "markets/new" => "markets#new", as: "new_market"
+
+ # only generate the specified routes
+ resources :markets
+ resources :vendors
+ resources :products
+ resources :sales
```

Comment:
Speaking for myself, I am still a little fuzzy on routing. We had just learned about the resource statements, so we implemented them, but I can see how they are redundant.